### ISSUE TITLE
LetEncrypt Certs not validated

### ISSUE THREAD

**Issue by guy-menlo** [rancher/rancher#18585]
**What kind of request is this (question/bug/enhancement/feature request):**

Bug

**Steps to reproduce (least amount of steps as possible):**

Create a new HA Rancher service. Create and associate a letsEncrypt Certificate. Verify the Certificate is working by checking on a browser https://rancher.example.com

Create a New Cluster and attempt to associate the node to this cluster..

time=“2019-03-04T09:04:18Z” level=info msg=“Connecting to proxy” url=“wss://rancher…/v3/connect/register”
time=“2019-03-04T09:04:18Z” level=error msg=“Failed to connect to proxy” error=“x509: certificate signed by unknown authority”
time=“2019-03-04T09:04:18Z” level=error msg=“Failed to connect to proxy” error=“x509: certificate signed by unknown authority”

So it looks like the container does have the LetEncrypt Certificates registered.

curl ‘https://rancher…/ping’
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.haxx.se/docs/sslcerts.html 1

I can fix this by adding the certificate to the cert store in /etc/ca-certificates.conf

Why aren’t the LetsEncrypt certs not installed by default??

**Result:**

Nodes will not join the cluster as the certificate is not seen as valid.

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
- Installation option (single install/HA):

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.4", GitCommit:"c27b913fddd1a6c480c229191a087698aa92f0b1", GitTreeState:"clean", BuildDate:"2019-02-28T13:37:52Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.6", GitCommit:"b1d75deca493a24a2f87eb1efde1a569e52fc8d9", GitTreeState:"clean", BuildDate:"2018-12-16T04:30:10Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:20:43 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:28:38 2018
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Comment by superseb** [comment]
Let's Encrypt uses intermediate which need to be present in the server certificate to provide complete chain up to the CA certificate. Please provide exact steps as requested to reproduce the situation (how did you get the certificates, what files did you use where, are you using cert-manager's Lets Encrypt or your own certificates?), also the Rancher version is missing from this issue.

**Comment by guy-menlo** [comment]
I was trying to use cert-manager, however that wasn't working for me for other reasons..I have an external tool which can generate LetsEncrypt certs via DNS.

I then used the Rancher UI to install the certificate crt and key to the "local->system".

Under workloads-> load balancing I then find the "rancher" ingress and assign the SSL cert to it.

I'm then able to connect to my rancher server via https://rancher....

On further digging around I've found that it's missing the cert "Let’s Encrypt Authority X3 (IdenTrust cross-signed)". As soon as I add that to the cert store /usr/share/ca-certificates/ and include into /etc/ca-certificates.conf. I can then issue, update-ca-certificates and it will start to work.
